### DISCUSSION TITLE
Windows.UI.Xaml.Unhandled exception occurs when show the popup in ContentDialog

### DISCUSSION THREAD

**Discussion by vijayarasan** [microsoft/microsoft-ui-xaml#7476] [Q&A]
I have loaded the popup in ContentDialog and show the popup by clicking the button Unhandled exception.

Here the popup code snippet in ContentDialog 

<ContentDialog

    x:Class="App1.ContentDialog1"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:App1"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d"
    Title="TITLE"
    PrimaryButtonText="Button1"
    SecondaryButtonText="Button2"
    PrimaryButtonClick="ContentDialog_PrimaryButtonClick"
    SecondaryButtonClick="ContentDialog_SecondaryButtonClick">
    <ContentDialog.Resources>
        <Style TargetType="local:PopUpHostExt">
            <Setter Property="Template">
                <Setter.Value>
                    <ControlTemplate TargetType="local:PopUpHostExt">
                        <Border>
                            <Popup VerticalOffset="10" HorizontalOffset="200"  x:Name="StandardPopup">
                                <ContentControl>
                                    <Border BorderBrush="{StaticResource ApplicationForegroundThemeBrush}"
                                    BorderThickness="2" Width="200" Height="300">
                                        <StackPanel HorizontalAlignment="Center" VerticalAlignment="Center">
                                            <TextBlock Text="Simple Popup" FontSize="16" Margin="10" HorizontalAlignment="Center" VerticalAlignment="Top"/>
                                            <CheckBox Margin="10" Content="CheckBox" IsChecked="True" HorizontalAlignment="Left"/>
                                            <Button Content="Close" Margin="10" HorizontalAlignment="Center"/>
                                        </StackPanel>
                                    </Border>
                                </ContentControl>
                            </Popup>
                        </Border>
                    </ControlTemplate>
                </Setter.Value>
            </Setter>
        </Style>
    </ContentDialog.Resources>
    <Grid x:Name="Output" HorizontalAlignment="Center" VerticalAlignment="Top" Grid.Row="1">
        <StackPanel x:Name="Stk_Panel">
            <Button Content="Show Popup" Margin="10" Click="ShowPopupOffsetClicked"/>
            <Border x:Name="PopUpPresenter"/>
        </StackPanel>
    </Grid>
</ContentDialog>

Replication procedure to reproduce the reported problem

Step 1: Run the Sample

Step 2: Click the Open ContentDialog button

Step 3: Click the Show Popup button

**Observed Behavior:**
Windows.UI.Xaml.Unhandled exception occurs when showing the popup in ContentDialog

**Expected Behavior:**
How to show the Popup properly without making an exception in UWP 

I have attached the sample for your reference,

**Sample Link:** [Sample.zip](https://github.com/microsoft/microsoft-ui-xaml/files/6610570/Sample.zip)

Can anyone share the idea to resolve the reported problem?
Regrads,
Vijayarasan S

**Comment by StephenLPeters** [comment]
It looks like you are setting IsOpen to true in the popup host's OnApplyTemplate method, which happens durring an arrange and measure pass. This isn't supported, you can either set the property to true in the xaml file or in the constructor, or wait for the loaded event to set the popup's isOpen property.

**Comment by vijayarasan** [comment]
Hi @StephenLPeters ,

Thanks for the update.

Sorry for the delay.

I have attached the sample to this problem. If possible can please modify the sample or provide a code snippet related to the solution for solving the problem. 

Regards,
Vijayarasan S

**Comment by StephenLPeters** [comment]
I think this would work:

```
    public class PopUpHostExt : ContentControl
    {
        Popup popUp;
        public PopUpHostExt()
        {
            this.DefaultStyleKey = typeof(PopUpHostExt);
            IsOpen = true;
            this.Loaded += PopUpHostExt_Loaded;
        }

        private void PopUpHostExt_Loaded(object sender, Windows.UI.Xaml.RoutedEventArgs e)
        {
            popUp = GetTemplateChild("StandardPopup") as Popup;
            popUp.IsOpen = true;
        }

        private bool isOpen;

        public bool IsOpen
        {
            get { return isOpen; }
            set 
            { 
                isOpen = value;
                if (popUp != null)
                    popUp.IsOpen = isOpen;

            }
        }

        protected override void OnApplyTemplate()
        {
            base.OnApplyTemplate();
        }
    }
```
